### PR TITLE
Parenthesize confusing conditional

### PR DESCRIPTION
**Pull request by andrewgaul**
Found via error-prone.

### CODE REVIEW

**Comment by andrewgaul** [comment]
@jawnclarke Please review.

File: dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/handlers/DimensionDataCloudControlErrorHandler.java [hunk 1]
```diff
@@ -49,9 +49,12 @@ public void handleError(HttpCommand command, HttpResponse response) {
...
                exception = new ResourceNotFoundException(message, exception);
-            } else if (message.contains("INVALID_INPUT_DATA") || message.contains("ORGANIZATION_NOT_VERIFIED")
-                  || message.contains("SYSTEM_ERROR") && !message.contains("RETRYABLE_SYSTEM_ERROR") || message
-                  .contains("CPU_SPEED_NOT_AVAILABLE") || message.contains("CONFIGURATION_NOT_SUPPORTED")) {
+            } else if ((message.contains("CONFIGURATION_NOT_SUPPORTED") ||
+                  message.contains("CPU_SPEED_NOT_AVAILABLE") ||
+                  message.contains("INVALID_INPUT_DATA") ||
+                  message.contains("ORGANIZATION_NOT_VERIFIED") ||
+                  message.contains("SYSTEM_ERROR")) &&
+                  !message.contains("RETRYABLE_SYSTEM_ERROR")) {
```
Comment (demobox) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/handlers/DimensionDataCloudControlErrorHandler.java]: @andrewgaul I'm not sure this is the same? As I read the original, it's

`... || ... || (message.contains("SYSTEM_ERROR") && !message.contains("RETRYABLE_SYSTEM_ERROR")) || ... || ...`

which seems to be different from the new version?

E.g. if `message.contains("INVALID_INPUT_DATA") == true` and `message.contains("RETRYABLE_SYSTEM_ERROR") == true`, the old version would evaluate to true but the new version would evaluate to false?

Comment (andrewgaul) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/handlers/DimensionDataCloudControlErrorHandler.java]: The intent is not clear to me.  @jawnclarke?

Comment (andrewgaul) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/handlers/DimensionDataCloudControlErrorHandler.java]: Reading this more closely, I agree with your interpretation.

**Pull request closed without merging**